examples/protocols/sntp/main/sntp_example_main.c

```diff
@@ -157,10 +162,33 @@ static void initialize_sntp(void)
 {
     ESP_LOGI(TAG, "Initializing SNTP");
     sntp_setoperatingmode(SNTP_OPMODE_POLL);
+
+/*
+ * If 'NTP over DHCP' is enabled, we set static ntp server address 'pool.ntp.org'
+ * as a secondary server. It will act as a fallback server in case that address
+ * provided via NTP over DHCP is not accessible
+ */
+#if defined LWIP_DHCP_GET_NTP_SRV && (SNTP_MAX_SERVERS > 1)
+    sntp_setservername(1, "pool.ntp.org");
+#else
     sntp_setservername(0, "pool.ntp.org");
+#endif
+
     sntp_set_time_sync_notification_cb(time_sync_notification_cb);
 #ifdef CONFIG_SNTP_TIME_SYNC_METHOD_SMOOTH
     sntp_set_sync_mode(SNTP_SYNC_MODE_SMOOTH);
 #endif
     sntp_init();
+
+    ESP_LOGI(TAG, "List of configured NTP servers:");
+
+    for (uint8_t i = 0; i < SNTP_MAX_SERVERS; ++i){
+        if (sntp_getservername(i)){
+            ESP_LOGI(TAG, "server %d: %s", i, sntp_getservername(i));
+        } else {
+            const ip_addr_t * _ip = sntp_getserver(i);
+            if (0 == _ip->type && _ip->u_addr.ip4.addr)    // non-empty IPv4 addresses
```

Comment:
```cpp
(0 == ip->type)
```
Would this compile if we disable `IPv6` ?

Reply:
Ugh, you are right, dual stack only. Actually I'm not sure what is the best way to handle it here. Assume we have IPv4 only NTPs and just skip otherwise? I'm not sure if LWIP is capable to do NTPoDHCP via IPv6, need test in a lab setup.
I can fix it for sure, just want to hear your suggestion?

Reply:
> I can fix it and limit to IPv4-only or dual stack

This sounds good. IDF supports only these two (dual stack and IPv4-only) settings, so that's perfectly okay to disregard IPv6-only scenario.

> LWIP is capable to do NTPoDHCP via IPv6,

lwip supports DHCPv6 in its stateless configuration and users can add options using `LWIP_HOOK_DHCP6_APPEND_OPTIONS` (so theoretically we can add `OPTION_NTP_SERVER (56)`) , but TBH I haven't tested that either, and even if we get this running it wouldn't be very practical to include it into this simple example.

This piece of code just prints addresses of NTP servers, correct? So not only those obtained by DHCP, but also the hardcoded ones, and maybe some of the names get resolved to IPv6 addresses (one day)? If you want to go the extra mile, you might look into how we can print the IP address in the IP-protocol agnostic way (using `ntoa()`-like API -- which again is not very readable if written in lwip)

Let's just assume we have IPv4 NTPs here. My only point was that if users disable `IPv6` in menuconfig this example won't compile.

Reply:
Thanks for your ideas, @david-cermak, appreciate it.

> This piece of code just prints addresses of NTP servers, correct?
it prints domain names for ntp's assigned statically and IP's for NTPs received via DHCP.

OK, I will concentrate on IPv4 and dual-stack case here and leave IPv6 opts for future improvements :)
Will update this PR and let you know than!
Cheers!

Reply:
Found a better approach. Now it works fine for IPv4-only and for dual-stack builds both for IPv4/IPv6 ntp addresses.

```
sntp_init: SNTP initialised
I (7400) example: List of configured NTP servers:
I (7410) example: server 0: 192.168.168.2
I (7410) example: server 1: pool.ntp.org
I (7420) example: server 2: 2A01:3F7::1
I (7420) example: Waiting for system time to be set... (1/15)
```

Reply:
Nice work, thanks for testing this!